<!DOCTYPE html>
<html>
<head>
<meta charset="utf-8">
<meta name="viewport" content="width=device-width, initial-scale=1.0"> 
<title>CCyCDB-definition</title>
<link rel="stylesheet" type="text/css" href="../styles.css"> 
<link rel="stylesheet" type="text/css" href="../df.css"> 
<style type="text/css">
h2 {
    font-size: 26px;
    color: #375D80;
    text-align: left;
    margin-left: 45px;
   margin-bottom: 0px;
    padding: 10px;
}

h3 {
    font-size: 20px;
    color: #375D80;
    text-align: left;
    margin-left: 45px;
    margin-top: 0px;
    margin-bottom: 0px;
    padding: 10px;
}

p{
    font-size: 16px;
    color: black;
    line-height:1.5;
    text-align: left;
    margin-left: 55px;
}
	
li {
  color: black;
  margin-left: 35px;
}

</style>
</head>

<body>
	<h1>CCycDB-Usage and Examples </h1>
	<hr>
    <h2>A guide to analyzing metagenomic data with CCycDB</h2>

	
	<!--  文件类型 -->
   <div>
  	<h3>Step 0. Prepare your files</h3>
	<p><strong>Input files</strong> supports 3 type of files</p>
	  
	  <ul>
		<li><p style=" margin-left: 5px;"><b>Read-based</b></p></li>
		<p>Raw reads (clean reads) or Merged Reads (Forward and reverse reads were merged into longer sequences by the program e.g. PEAR)</p>
	
		<li><p style=" margin-left: 5px;"><b>Assembly-based</b></p></li>
		<p>Contigs generated through metagenome assmebly by (e.g., MEGAHIT, MetaSPAdes, SPAdes).</p>	
		<li><p style=" margin-left: 5px;"><b>Tabular Files</b></p></li>
		<p>BLAST table (delimited with "\t") generated through sequences similarity searching tools (e.g., BLAST, USEARCH, DIAMOND) </p>
		
	</ul>
	</div>
	
	<!--  数据库下载 -->
	<div>
	<h3>Step 1. Download </h3>
	  <p>xx</p>
	  <p>xx</p>		
	</div>

	<!--  UASGE -->
	<div>
	<h3>Step 2. Annotation </h3>
	  <p class="text">xxx</p>
	  <p>xx</p>
	  <p class="text"><b>[Options:]</b></p>
	</div>
	<!-- 参数 -->
	<div class="graybox">
		<table class="table table-hover pt-2">
			<tbody>
			  <tr><td class="td-left">-situation</td>
				<td class="td-right">the situation for input files (read-based|assembly-based|tabular)</td>
			  </tr>
			  <tr>
				<td class="td-left">-wd</td>
				<td class="td-right">The directory may or may not exist.</td>
			  </tr>
			  <tr>
				<td class="td-left">-od</td>
				<td class="td-right">File type. Specify the extensions of your input files (fastq, fastq.gz, fasta, fasta.gz, fq, fq.gz, fa, fa.gz) or (faa, fna). <br>Please make sure that the filetype is correct according to the tool selected by -m. <br>E.g., if -m usearch, the filetype supports (fastq|fasta); if -m blast, the filetype supports fasta.</td>
			  </tr>
			  <tr>
				<td class="td-left">-m</td>
				<td class="td-right">Database searching program you plan to use (diamond|usearch|blast)</td>
			  </tr>
			  <tr>
				<td class="td-left">-f</td>
				<td class="td-right">Specify the extensions of your sequence files (fastq, fastq.gz, fasta, fasta.gz, fq, fq.gz, fa, fa.gz) or (faa, fna). <br>Please make sure that the filetype is correct according to the tool selected by -m. <br>E.g., if -m usearch, the filetype supports (fastq|fasta); if -m blast, the filetype supports fasta.</td>
			  </tr>
			  <tr>
				<td class="td-left">-s</td>
				<td class="td-right">Sequence type (nucl|prot)</td>
			  </tr>
			  <tr>
				<td class="td-left">-tpm</td>
				<td class="td-right">(0|1) (default: 0). "1" need $sample.tpm exist in the work directory</td>
			  </tr>
			  <tr>
				<td class="td-left">-norm</td>
				<td class="td-right">(0|1) 0: don`t need random sampling; 1: need random sampling.[-situation assembly-based] is a prerequisite for this parameter</td>
			  </tr>
			  <tr>
				<td class="td-left">-rs</td>
				<td class="td-right">the number of sequences for random subsampling. (default: the lowest number of sequences).<br>
					Note: [-norm 1] is a prerequisite for this parameter.</td>
			  </tr>
			  <tr>
				<td class="td-left">-thread</td>
				<td class="td-right">number of threads (default: 2)</td>
			  </tr>
			</tbody>
		</table>
	</div>
	

	<div >
	  <p style="text-align: left; margin-left: 55px;font-size: 18px;"> <b>Depending on the tools used, you may want to cite also:</b></p>
	  <p style="text-indent: 10px;">DIAMOND:  Buchfink B, Xie C, Huson D H. Fast and sensitive protein alignment using DIAMOND[J]. Nature methods, 2015, 12(1): 59-60.</p>
	  <p style="text-indent: 10px;">BLASTX: Boratyn G M, Camacho C, Cooper P S, et al. BLAST: a more efficient report with usability improvements[J]. Nucleic acids research, 2013, 41(W1): W29-W33.</p>
	 <p style="text-indent: 10px;">USEARCH:  Edgar R C. Search and clustering orders of magnitude faster than BLAST[J]. Bioinformatics, 2010, 26(19): 2460-2461.</p>
	 <p style="text-indent: 10px;">CSVTK:  Csvtk—CSV/TSV Toolkit. Available online: https://bioinf.shenwei.me/csvtk/</p>
	 <p style="text-indent: 10px;">SEQKIT:  Shen W, Le S, Li Y, et al. SeqKit: a cross-platform and ultrafast toolkit for FASTA/Q file manipulation[J]. PloS one, 2016, 11(10): e0163962.</p>
	</div>
	
</body>
</html>
